pkg/manifests/manifests_test.go

```diff
@@ -276,6 +278,131 @@ func TestUnconfiguredManifests(t *testing.T) {
 	}
 }
 
+func TestHTTPConfig(t *testing.T) {
+	type checkFunc func(*monv1.Alertmanager) error
+
+	checks := func(cs ...checkFunc) checkFunc {
+		return func(a *monv1.Alertmanager) error {
+			for _, f := range cs {
+				if err := f(a); err != nil {
+					return err
+				}
+			}
+			return nil
+		}
+	}
+
+	hasContainers := func(expected bool) checkFunc {
+		return func(a *monv1.Alertmanager) error {
+			if got := len(a.Spec.Containers) > 0; got != expected {
+				return fmt.Errorf("expected spec to have containers %t, got %t", expected, got)
```

Comment:
small doubt: what would it mean for the error to say `expected spec to have containers false`? (or `true` for that matter)

Reply:
yes, that would be the meaning ... hmm, do you have a better wording? :thinking:

Reply:
blah nothing short comes to mind. we could make it a bit more verbose and interpretable, like:
```go
var negative string
if !expected {
    negative = "not"
}
fmt.Errorf(strings.Join(" ", "expected spec", negative, "to have containers, got %d containers"), len(a.Spec.Containers))
```

But this is not necessary IMO

Reply:
suggestion: as I am never asserting `hasContainers(false)` I can omit the parameter, always expecting `true`, which makes it more reasonable for now. Let's go with that.

```
	hasContainers := func() checkFunc {
		return func(a *monv1.Alertmanager) error {
			if len(a.Spec.Containers) == 0 {
				return errors.New("expected spec to have containers, but it doesn't have any")
			}
			return nil
		}
	}
```